### DISCUSSION TITLE
Option to print anonymous nodes with tree-sitter cli

### DISCUSSION THREAD

**Discussion by leotrs** [tree-sitter/tree-sitter#1986] [Ideas] [answered]
Currently `tree-sitter parse` only outputs the named nodes. It would be useful for debugging to have the option to also print anonymous nodes.

When using the -x flag, it will include the anonymous nodes in the XML output. However, this doesn't include the start/end point numbers.

**Comment by ahlinc** [comment]
Exactly the same functionality currently exists in a fork that eventually will be upstreamed. The fork can be installed with:
`cargo install --git https://github.com/ahlinc/tree-sitter --branch alpha`, use an `-oc` flag. More info about it exist in a Discord server, a link to it pinned in discussions.

**Reply by leotrs** [reply]
Great! Looking forward to seeing that merge.

**Comment by lollipopman** [comment]
Looks like this patch was never merged in, would still be a nice feature to have!

**Reply by WillLillis** [reply, marked as answer]
This was added in https://github.com/tree-sitter/tree-sitter/pull/3762. 

Give `tree-sitter parse --cst` a try!

**Reply by lollipopman** [reply]
Thanks @WillLillis, However, I just tried `--cst` with 0.25.6, but I don't see anonymous nodes?

**Reply by WillLillis** [reply]
I'd have to double check to see if every last anonymous node was included in the cst output, but I am confident that the current `--cst` implementation matches ahlinc's original `-oc` in the fork. For example, the semicolon is an anonymous node:
<img width="852" height="233" alt="image" src="https://github.com/user-attachments/assets/e6e95289-ed97-4a88-9ce3-bf1b536373ff" />

**Reply by lollipopman** [reply]
Thanks, but I was hoping for the names of anonymous nodes, e.g. `$._heredoc_body`
